lib/MusicBrainz/Server/Controller/Admin/Attributes.pm

```diff
@@ -93,12 +95,22 @@ sub edit : Chained('attribute_base') Args(1) RequireAuth(account_admin) {
     my $form = $c->form( form => $form_name, init_object => $attr );
 
     if ($c->form_posted && $form->process( params => $c->req->params )) {
-        $c->model('MB')->with_transaction(sub {
-            $c->model($model)->update($id, { map { $_->name => $_->value } $form->edit_fields });
-        });
-
-        $c->response->redirect($c->uri_for('/admin/attributes', $model));
-        $c->detach;
+        my $old_entity_type = $attr->entity_type;
+        my $new_entity_type = $form->field("entity_type")->value;
+
+        if ($old_entity_type ne $new_entity_type) {
+            $form->field('entity_type')->add_error(
+                l('Entity type \'{entity_type}\' cannot be changed; please delete and readd attribute instead.',
```

Comment:
Done in https://github.com/metabrainz/musicbrainz-server/pull/471/commits/73daae629f043dae0ee38d7e12867402c91cb093.
On a side note: I disabled the `<select>` element using JS because TT seems to ignore `disabled` attribute.